### DISCUSSION TITLE
How frequently does the GraphQL API update?

### DISCUSSION THREAD

**Discussion by howitworks18** [snapshot-labs/snapshot#2992] [General]
When casting a vote on a proposal, the single proposal API response doesn't get updated immediately and the "scores_updated" in the response doesn't seem to have a  predictable cadence.

**Comment by metrox-eth** [comment]
Hey, that's correct there is a 10 min lag when querying an active proposal using GraphQL, to fetch fresh data you must use snapshot.js. https://docs.snapshot.org/snapshot.js

**Reply by howitworks18** [reply]
im looking at the snapshot.js library, which one of these fetches the proposal? The docs for the library are pretty sparse. 

![Screen Shot 2022-08-15 at 4 17 45 PM](https://user-images.githubusercontent.com/20389993/184720247-b00d600c-b273-492b-9bbe-a71afbd835cb.png)

**Reply by metrox-eth** [reply]
You were talking of votes in your original message not proposal right ? You can fetch proposals from the API and votes from snapshot.js

**Reply by metrox-eth** [reply]
@ChaituVR am I right ?

**Reply by bonustrack** [reply]
> im looking at the snapshot.js library, which one of these fetches the proposal? The docs for the library are pretty sparse.
> 
> ![Screen Shot 2022-08-15 at 4 17 45 PM](https://user-images.githubusercontent.com/20389993/184720247-b00d600c-b273-492b-9bbe-a71afbd835cb.png)

You can query all the data, proposals, votes, spaces etc within the GraphQL API, you can use any GraphQL library to do the requests, otherwise on Snapshot.js we have a function `subgraphRequest` that would work too.

**Reply by howitworks18** [reply]
> You were talking of votes in your original message not proposal right ? You can fetch proposals from the API and votes from snapshot.js

i thought for some reason the current vote tally would be in the proposal response like it is with GraphQL. There's an end point for the current vote tally in snapshot.js?

**Reply by metrox-eth** [reply]
Ho, I misunderstood, though you was referring to individual votes, not results.

**Reply by howitworks18** [reply]
sorry, for clarity I'm working on a solution that takes the voting experience off snapshot.org and onto a site I'm working on. I'm using the API to get proposals, voting power, and the library to cast votes. I was using the API to get the latest vote tally after new votes have been added but the 10min delay wasn't ideal.

**Reply by metrox-eth** [reply]
Gotcha, the catch here is that until we have delegation as a core feature we need to have a scheduled calculation every 10min or so. Before it was calculated on the fly each time a proposal was loaded and was too resources intensive / not scalable.
However things should improve overtime, core delegation is one of the next in the priority list.

**Comment by howitworks18** [comment]
ok cool, thanks for the responses! I'll probably do some sort of solution with firebase, for now, that keeps track of the vote counts in real-time and syncs with the API.

**Reply by metrox-eth** [reply]
Yes, if you build a front that is supposed to withstand the load of a few thousand votes, it will work. We needed these changes when we had 300,000 votes in one weekend.

**Comment by VitalkaaNX** [comment]
Gooood

**Comment by Sorikovich** [comment]
thx
